x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/index.tsx

```diff
@@ -286,10 +287,10 @@ const RuleDetailsPageComponent = () => {
 
   const alertDefaultFilters = useMemo(
     () => [
-      ...(ruleId != null ? buildAlertsRuleIdFilter(ruleId) : []),
+      ...(ruleId != null ? buildAlertsRuleIdFilter(ruleId, showThreatMatchesOnly) : []),
```

Comment:
@ecezalp here's what I was imagining for this page:

```suggestion
      ...(ruleId != null ? buildAlertsRuleIdFilter(ruleId) : []),
      ...buildThreatMatchFilter(showThreatMatchesOnly),
```

and in that way we could keep the filtering functions' logic separated